code/player/Inventory.Perks.cs

```diff
@@ -0,0 +1,83 @@
+using System.Collections.Generic;
+
+using Sandbox;
+
+using TTTReborn.Items;
+
+namespace TTTReborn.Player
+{
+    public partial class PerksInventory
+    {
+        private List<TTTPerk> PerkList { get; set; } = new();
+        private Inventory Inventory;
+
+        public PerksInventory(Inventory inventory) : base()
+        {
+            Inventory = inventory;
+        }
+
+        public bool Give(TTTPerk perk)
+        {
+            if (Has(perk))
+            {
+                return false;
+            }
+
+            PerkList.Add(perk);
+
+            TTTPlayer player = Inventory.Owner as TTTPlayer;
+
+            perk.Equip(player);
+
+            if (Host.IsServer)
+            {
+                player.ClientAddPerk(To.Single(player), perk.Name);
+            }
+
+            return true;
+        }
+
+        public bool Take(TTTPerk perk)
+        {
+            if (!Has(perk))
+            {
+                return false;
```

Comment:
Just a minor thing to consider, I see "issues" making some of these methods return a `bool` since we don't exactly know what the return type means. 

For example, I think most would assume that `Take` would return `True` if the player no longer has the perk on them. However, in this case, we return `False` if the perk removal fails.

Something to consider.